### PR TITLE
Yuval/comments for patch 19

### PR DESCRIPTION
**Pull request by yuval-lb**
21.1 comments

### CODE REVIEW

File: cinder/volume/drivers/lightos.py [hunk 1]
```diff
@@ -53,7 +53,7 @@
                 default='443',
                 help='The TCP/IP port at which the LightOS API'
                      ' endpoints listen.'
-                     ' Port 443 is used for HTTPS other values'
+                     ' Port 443 is used for HTTPS and other values'
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:56]: +1

File: cinder/volume/drivers/lightos.py [hunk 2]
```diff
@@ -363,11 +363,11 @@ def __init__(self, *args, **kwargs):
...
         if not self.configuration.__dict__.get("initiator_connector", None):
-            self.configuration.initiator_connector = \
-                "os_brick.initiator.connector.InitiatorConnector"
+            self.configuration.initiator_connector = (
+                "os_brick.initiator.connector.InitiatorConnector")
         if not self.configuration.__dict__.get("lightos_client", None):
-            self.configuration.lightos_client = \
-                "cinder.volume.drivers.lightos.LightOSConnection"
+            self.configuration.lightos_client = (
+                "cinder.volume.drivers.lightos.LightOSConnection")
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:370]: +1

File: cinder/volume/drivers/lightos.py [hunk 3]
```diff
@@ -420,9 +420,9 @@ def create_cloned_volume(self, volume, src_vref):
             self._create_volume(volume,
                                 src_snapshot_lightos_name=snapshot_name)
         except Exception as e:
-            LOG.warn("Failed to create volume %s from intermediate "
-                     " snapshot %s. Trying to clean up.",
-                     src_volume_name, snapshot_name)
+            LOG.error("Failed to create volume %s from intermediate "
+                      " snapshot %s. Trying to clean up.",
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:424]: note, not important but there is white space both after intermediate and before snapshot

File: cinder/volume/drivers/lightos.py [hunk 4]
```diff
@@ -608,7 +608,7 @@ def _get_lightos_uuid(self, project_name, volume):
 
         lightos_uuid = data.get('UUID')
         if not lightos_uuid:
-            LOG.error('Failed to get LightOS volume UUID',
+            LOG.error('Failed to get LightOS volume UUID status: %s, data: %s',
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:611]: +1

File: cinder/volume/drivers/lightos.py [hunk 5]
```diff
@@ -674,9 +674,11 @@ def _create_volume(self, volume, src_snapshot_lightos_name):
                 self._delete_lightos_volume(project_name, lightos_uuid)
                 # wait for openstack to call us again to create it
 
-        msg = 'Did not succeed creating LightOS volume with UUID %s \
-        status_code %s last state %s' % \
-              (lightos_uuid, status_code, vol_state)
+        msg = (
+            "Did not succeed creating LightOS volume with UUID %(uuid)s"
+            " status_code %(code)s last state %(state)s" %
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:679]: +1

File: cinder/volume/drivers/lightos.py [hunk 6]
```diff
@@ -766,7 +766,7 @@ def _delete_lightos_volume(self, project_name, lightos_uuid):
             if status_code == httpstatus.OK:
                 break
 
-            LOG.info(
+            LOG.warn(
```
Comment (ronen-lb) [inline cinder/volume/drivers/lightos.py:769]: not critical, but suugest to add to returned string "retrying operation.
i.e:
"delete_volume for volume with LightOS UUID %s failed \
￼                with status code %s response %s, retrying operation"
so it is clearer why this is a warning and not error.

also note, post merge,   not sure if we want need this retry, as we also have a retry loop within send_cmd,
additionally lightOS may return error not_found. in which case it is redundent to retry error.

we need to look at the retry logic further, also possibly in lightOS need to make delete volume idempotent (i.e if voule already deleted no error)

**Review by ronen-lb — APPROVED** [review]

File: doc/source/configuration/block-storage/drivers/lightbits-lightos-driver.rst [hunk 7]
```diff
@@ -25,7 +25,6 @@ Supported operations
 - Delete snapshot
 - Create volume from snapshot
 - Create volume from volume (clone)
-- Extend a volume
```
Comment (ronen-lb) [inline doc/source/configuration/block-storage/drivers/lightbits-lightos-driver.rst:28]: +1

**Comment by muliby-lb** [comment]
@yuval-lb is this still needed? Merge/close?

**Pull request closed without merging**